odoo/addons/base/models/res_lang.py

```diff
@@ -159,6 +159,17 @@ def _activate_lang(self, code):
             lang.active = True
         return lang
 
+    def _activate_and_update_lang(self, code):
+        """ Activate languages and update their translations
+        :param code: code of the language to activate
+        :return: the language matching 'code' activated
+        """
+        lang = self._activate_lang(code)
+        if lang:
+            mods = self.env['ir.module.module'].search([('state', '=', 'installed')])
+            mods._update_translations(code)
+        return lang
```

Comment:
There's nothing to do if the language is installed already.  So I would mimic what method `_activate_lang()` does, and reuse method `toggle_active()`, which I guess you copied the code from.
```suggestion
        lang = self.with_context(active_test=False).search([('code', '=', code)])
        if lang and not lang.active:
            lang.toggle_active()
        return lang
```